### DISCUSSION TITLE
[Angular] Use checkboxes for an input of type Set

### DISCUSSION THREAD

**Discussion by c-harding** [storybookjs/storybook#27157] [Help]
### Summary

I have got a component that takes a Set&lt;T&gt; as input.

```ts
export class SetsComponent {
  readonly inputs = input.required<Set<MyEnum>>();
}
```

By default this is shown as a string input in Storybook, but I would like to use a control such as 'inline-check'.
![image](https://github.com/storybookjs/storybook/assets/8607022/8604e778-52f2-4543-93f1-cbc6f81497a8)

```ts
export default {
  ...
  argTypes: {
    inputs: {
      options: [MyEnum.A, MyEnum.B],
      control: { type: 'check' },
    },
  },
}
```

This causes the error following error:

```
Uncaught TypeError: e.includes is not a function
```

Apparently the options are treated as an array rather than a Set. How can I use Sets instead, e.g. is there a way to map a prop into a form usable by the control and vice versa?

### Additional information

For the reproduction, I have two almost identical components: SetsComponent and ArraysComponent. ArraysComponent works as expected, and I would like SetsComponent to work in the same way in Storybook

### Create a reproduction

https://stackblitz.com/edit/github-m5gfbn?file=src%2Fstories%2Fsets.stories.ts

**Comment by Marklb** [comment]
This doesn't seem to be limited to Angular. I tried to convert your stories to a React equivalent and the only difference was that where you got a string input, React showed an Object input. I think it is just the docs addon isn't handling `Set` type args.

I don't know is there is a good solution, at the moment, but if you just want the functionality and don't mind the code being odd then the following could work. It is not something I really think is a good solution though.

```ts
export const INLINE_CHECK: Story = {
  render: args => ({
    props: {
      ...args,
      inputs: new Set(args.inputs) // Manually mapping the array to a set.
    },
  }),
  args: {
    // inputs: new Set([MyEnum.A]),
    inputs: [MyEnum.A] as any, // Using an array, even though it is the wrong type.
  },
  argTypes: {
    inputs: {
      options: [MyEnum.A, MyEnum.B],
      control: { type: 'inline-check' },
    },
  },
};
```
